play/src/hooks/useFetchAndSubscribeGameSession.tsx

```diff
@@ -28,38 +34,42 @@ export default function useFetchAndSubscribeGameSession(
       setError('');
     }
     // if gamesessionId is null, set error and prevent api calls
-    if (isNullOrUndefined(gameSessionId)){
+    if (isNullOrUndefined(gameSessionId)) {
       setError(`${t('error.connecting.gamesessionerror')}`);
       setIsLoading(false);
-    }
-    else {
+    } else {
       apiClient
         .getGameSession(gameSessionId)
         .then((fetchedGame) => {
           if (!fetchedGame)
             setError(`${t('error.connecting.gamesessionerror')}`);
+          if (!ignore) setGameSession(fetchedGame);
           try {
-            const gameSessionSubscription = apiClient.subscribeUpdateGameSession(
-              fetchedGame.id,
-              (response) => {
-                if (!response)
-                  setError(`${t('error.connecting.subscriptionerror')}`);
-                // Update the gameSession object and trigger the callback
-                if (!ignore)
-                  setGameSession((prevGame) => ({ ...prevGame, ...response }));
-              }
-            );
+            const gameSessionSubscription =
+              apiClient.subscribeUpdateGameSession(
+                fetchedGame.id,
+                (response) => {
+                  if (!response)
+                    setError(`${t('error.connecting.subscriptionerror')}`);
+                  // Update the gameSession object and trigger the callback
+                  if (!ignore) {
+                    setGameSession((prevGame) => ({
+                      ...prevGame,
+                      ...response,
+                    }));
+                    setIsRejoin(false);
```

Comment:
should probably be the first statement here

Reply:
fixed